src/mono/mono/metadata/marshal-lightweight.c

```diff
@@ -2095,11 +2095,12 @@ emit_delegate_invoke_internal_ilgen (MonoMethodBuilder *mb, MonoMethodSignature
 					mono_mb_emit_ldarg (mb, i + 1);
```

Comment:
This optimized version depends on target_method as well, so it might need to be removed.

Reply:
I mean that optimized version needs to be removed since it will make the wrapper depend on target_method.